platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java

```diff
@@ -42,8 +44,28 @@
 public final class DataFileReader<D> implements AutoCloseable, Comparable<DataFileReader<D>>, IndexedObject {
     /** FileChannel's for each thread */
     private static final ThreadLocal<ByteBuffer> BUFFER_CACHE = new ThreadLocal<>();
-    /** FileChannel's for each thread */
-    private final FileChannel fileChannel;
+    /** Max number of file channels to use for reading */
+    private static final int MAX_FILE_CHANNELS = 8;
+    /**
+     * When a data file reader is created, a single file channel is open to read data from the
+     * file. This channel is used by all threads. Number of threads currently reading data is
+     * tracked in {@link #fileChannelsInUse}. When the number of threads per opened file channel
+     * exceeds this threshold, a new file channel is open, unless there are {@link #MAX_FILE_CHANNELS}
+     * channels are already opened.
+     */
+    private static final int THREADS_PER_FILECHANNEL = 3;
```

Comment:
Just curious, how did you come up with this value? Why is it 3 and not 5, let's say?

Reply:
Also, would it be too wasteful to have a channel per thread?

Reply:
@jsync-swirlds thank you for the clarification, it helps. The question about `THREADS_PER_FILECHANNEL` still stands, though.

Reply:
As discussed in slack, I'll increase it from 3 to 8. Changing to fully asynchronous mode is tracked in a separate epic, it's much more work (and design thoughts) than this simple PR.